pkg/util/containers/v2/metrics/containerd/collector.go

```diff
@@ -83,37 +81,42 @@ func (c *containerdCollector) GetContainerStats(containerID string, cacheValidit
 		return nil, err
 	}
 
-	switch metricsVal := metrics.(type) {
-	case *v1.Metrics:
-		container, err := c.client.Container(containerID)
-		if err != nil {
-			return nil, fmt.Errorf("could not get container with ID %s: %s", containerID, err)
-		}
+	if winStats, ok := metrics.(*wstats.Statistics); ok {
+		windowsMetrics := winStats.GetWindows()
 
-		OCISpec, err := c.client.Spec(container)
-		if err != nil {
-			return nil, fmt.Errorf("could not retrieve OCI Spec from container with ID %s: %s", containerID, err)
+		if windowsMetrics == nil {
+			return nil, fmt.Errorf("error getting Windows metrics for container with ID: %s: %s", containerID, err)
 		}
 
-		info, err := c.client.Info(container)
-		if err != nil {
-			return nil, fmt.Errorf("could not retrieve the metadata of the container with ID %s: %s", containerID, err)
-		}
+		return getContainerdStatsWindows(windowsMetrics), nil
+	}
 
-		processes, err := c.client.TaskPids(container)
-		if err != nil {
-			return nil, fmt.Errorf("could not retrieve the processes of the container with ID %s: %s", containerID, err)
-		}
+	container, err := c.client.Container(containerID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get container with ID %s: %s", containerID, err)
```

Comment:
Would it make sense to use `%w` format specifier for the error here? (and other places where the error is wrapped)